src/openrct2-ui/windows/EditorInventionsList.cpp

```diff
@@ -305,10 +305,17 @@ static rct_research_item *window_editor_inventions_list_get_item_from_scroll_y(s
         researchItem++;
     }
 
-    for (; researchItem->rawValue != RESEARCHED_ITEMS_SEPARATOR && researchItem->rawValue != RESEARCHED_ITEMS_END; researchItem++) {
+    for (; researchItem->rawValue != RESEARCHED_ITEMS_SEPARATOR && researchItem->rawValue != RESEARCHED_ITEMS_END; researchItem++)
+    {
         y -= SCROLLABLE_ROW_HEIGHT;
         if (y < 0)
+        {
+            if (research_item_is_always_researched(researchItem))
```

Comment:
It'd be nicer to perform this check only where it's needed. It would make sense for `window_editor_inventions_list_get_item_from_scroll_y` to return the research item for always researched items too.

Reply:
but this is exactly what fixes [this behaviour](https://github.com/OpenRCT2/OpenRCT2/issues/7601#issuecomment-393935561)

Reply:
This will be a double-edged sword:

I just moved the check to the calling functions.  
I then recognized, that the check is required in every function that calls either  
`window_editor_inventions_list_get_item_from_scroll_y_include_seps` or  
`window_editor_inventions_list_get_item_from_scroll_y`

So apparently it's always required and should stay in  
`window_editor_inventions_list_get_item_from_scroll_y`  
so that new calls to it don't introduce bugs with always-researched items.

We could either risk introduction of bugs by not checking the returnvalue of the `_get_item_from` functions against `is_always_researched` or risk bugs by not getting always-researched items.

The second one can be mitigated with renaming the getter functions

Reply:
It's not always required. Look at `window_editor_inventions_list_scrollmouseover` for example. Right now, because of the check, you don't see the row highlighted when hovering over an always-invented research item. The only place where the check is needed is on `window_editor_inventions_list_scrollmousedown` (for making it non-pickable) and `window_editor_inventions_list_cursor` (for setting the hand cursor).

All other calls to these functions (and `get_research_item_at` which is basically a wrapper) don't need this check. I've tested this locally now, to make extra sure they really aren't needed. If you're okay with it, I can push my changes here.

Reply:
I'd say it should not  be highlighted etc as well

Reply:
It's how the original games does it too, but it can't hurt to have the missing highlight as extra user feedback. To fic the cursor feedback, the check should be removed from `window_editor_inventions_list_cursor` though.

Reply:
what about `window_editor_inventions_list_drag_moved`. putting it in there would prevent dropping items between the non-pickables

Reply:
Sounds like a nice thing to do. Maybe add comments too, to explain why some of the checks are there.